v2/client.go

```diff
@@ -102,6 +113,13 @@ func NewHTTPClient(conf HTTPConfig) (Client, error) {
 		return nil, errors.New(m)
 	}
 
+	switch conf.WriteEncoding {
+	case DefaultEncoding:
+	case GzipEncoding:
+	default:
+		return nil, errors.New(fmt.Sprintf("Unsupported encoding : %s", conf.WriteEncoding))
```

Comment:
The Go standard is to start all errors with lowercase letters and the space before the colon isn't necessary. Also, instead of using `errors.New(fmt.Sprintf(msg))` you should just use `fmt.Errorf`.

Reply:
fixed